### PR TITLE
Additional check on flag ignore_non_existent_counters if the PdhExpandWildCardPathW returns no errors

### PR DESCRIPTION
**Pull request by narph**
Additional check on flag `ignore_non_existent_counters` if the `PdhExpandWildCardPathW` returns no errors.
The flag is checked only if the `PdhExpandWildCardPathW` returns an error. There could be cases where this func does not return an error but will not expand the counter path or return any counter paths.
An extra check is added in this cases.

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -270,6 +270,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Fix regular expression to detect instance name in perfmon metricset. {issue}14273[14273] {pull}14666[14666]
 - Vshpere module splits `virtualmachine.host` into `virtualmachine.host.id` and `virtualmachine.host.hostname`. {issue}7187[7187] {pull}7213[7213]
 - Fixed bug with `elasticsearch/cluster_stats` metricset not recording license ID in the correct field. {pull}14592[14592]
+- Add extra check on `ignore_non_existent_counters` flag if the PdhExpandWildCardPathW returns no errors. {pull}14797[14797]
```
Comment (adriansr) [inline CHANGELOG.next.asciidoc]: can you add some context here so that users know that this belongs to the `windows/perfmon` metricset

Comment (narph) [reply CHANGELOG.next.asciidoc]: done, thank you.

**Comment by ph** [comment]
Added the need backport label, 7.x and 7.5 seems to be appropriate target for this fix.

**Comment by narph** [comment]
jenkins test this

**Pull request merged**